Logger/BaseLoggerMixins.cs

```diff
@@ -1,7 +1,43 @@
-﻿namespace Logger
+﻿using System;
+
+namespace Logger;
+
+public static class BaseLoggerMixins
 {
-    public static class BaseLoggerMixins
+    public static void Error(this BaseLogger logger, string message, params object[] args)
     {
-        
+        if (logger is null)
+        {
+            throw new ArgumentNullException(nameof(logger));
```

Comment:
```suggestion
            throw new ArgumentNullException(nameof(logger));
```
I think you left a comment about not understanding the use of nameof, however, it appears that you are using it correctly here. Is there something I can clarify?

Reply:
No, it's all good. Below the PR's there are collapsed line-specific conversations I resolved once I received clarification.